File: server/src/main/java/com/example/server/dto/User/UserDto.java
package com.example.server.dto.User;

import com.example.server.dto.Album.AlbumSimpleDto;
import com.example.server.dto.Author.AuthorDto;
import com.example.server.dto.Playlist.PlaylistSimpleDto;
import lombok.Data;

import java.util.Set;

@Data
public class UserDto
{
    private String id;
    private String username;
    private String email;
    private String imgUrl;
    private Set<AlbumSimpleDto> favouriteAlbums;
    private Set<PlaylistSimpleDto> favouritePlaylists;
    private AuthorDto authorDto = null;
}

File: client/src/models/UserDto.ts
import type { PlaylistSimpleDto } from "./DTO/PlaylistSimpleDto.ts";
import type { AlbumSimpleDto } from "./DTO/album/AlbumSimpleDto.ts";
import type {AuthorDto} from "./AuthorDto.ts";

export interface UserDto {
  id: string;
  username: string;
  email: string;
  avatarUrl?: string;
  subscriptionsCount: number;
  favoriteAlbums: AlbumSimpleDto[];
  favoritePlaylists: PlaylistSimpleDto[];
  authorDto?: AuthorDto;
}
